docs/_future/custom_functions.md

```diff
@@ -0,0 +1,217 @@
+---
+title: Custom functions
+order: 5
+slug: custom-functions
+lang: en
+layout: tutorial
+permalink: /future/custom-functions
+---
+# Custom functions <span class="label-beta">BETA</span>
+
+<div class="section-content">
+On the next-generation platform, you can build **custom functions**, reusable building blocks of automation that are deployed to our infrastructure and accept inputs, perform some calculations, and provide outputs. Functions can be used as steps in [Workflows](/bolt-js/future/workflows)&mdash;and Workflows are invoked by [Triggers](/bolt-js/future/triggers).
+
+To create a function, we need to do the following: 
+- [define the function](#define) in the Manifest;
+- [implement the function](#implement) in its respective source file;
+</div>
+
+---
+
+### Defining a function {#define}
+Functions are defined in your app via the `DefineFunction` method, which is part of the SDK that gets included with every newly created project.
+
+Let's take a look at a simple function:
+```js
```

Comment:
@srajiang a `source_file` property was mentioned here as "required" but i removed it because it seemed like we didn't need it. thoughts on adding it back in for the bolt context?

Reply:
Yep source_file is only relevant for building and deploying to slack!